### PR TITLE
update y axis label based on unit type

### PR DESCRIPTION
**Pull request by cshanahan1**
Check unit type (e.g flux density, surface brightness, counts, etc) for generating display label for the y axis in spectral viewer. Previously it was hard coded to always display 'flux density' no matter the input unit.

Closes https://github.com/spacetelescope/jdaviz/issues/2701

### CODE REVIEW

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 1]
```diff
@@ -545,7 +545,22 @@ def _plot_uncertainties(self):
...
-        # Set axes labels for the spectrum viewer
+        # Set y axes labels for the spectrum viewer
+        if y_unit.is_equivalent(u.Jy):
+            flux_unit_type = "Flux density"
+        elif y_unit.is_equivalent(u.Jy / u.sr):
+            flux_unit_type = "Surface brightness"
+        elif y_unit.is_equivalent(u.erg / (u.s * u.cm**2 * u.Hz)):
+            flux_unit_type = 'Flux'
+        elif y_unit.is_equivalent(u.e / u.s):
+            flux_unit_type = "Counts"
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:556]: "counts" is very ambiguous by STScI standard. I don't think this is correct.

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 2]
```diff
@@ -545,7 +545,22 @@ def _plot_uncertainties(self):
                 self.figure.marks = list(self.figure.marks) + [error_line_mark]
 
     def set_plot_axes(self):
-        # Set axes labels for the spectrum viewer
+        # Set y axes labels for the spectrum viewer
+        if y_unit.is_equivalent(u.Jy):
+            flux_unit_type = "Flux density"
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:550]: Besides "surface brightness" that Kyle is working on, I think the rest could be inferred from `y_unit.physical_type`?

Comment (kecnry) [reply jdaviz/configs/specviz/plugins/viewers.py:550]: re surface brightness, see https://github.com/astropy/astropy/pull/16032

Comment (cshanahan1) [reply jdaviz/configs/specviz/plugins/viewers.py]: I can still do it this way, but some of the labels we want will differ from what is returned by `physical_type` ( for example, an equivalency u.W should make the axis label 'luminosity', and u.electron/u.s should be 'counts') so it might not actually be less lines

File: CHANGES.rst [hunk 3]
```diff
@@ -98,6 +98,9 @@ Mosviz
 
 Specviz
 ^^^^^^^
+- Check unit type (e.g flux density, surface brightness, counts, etc) for generating
```
Comment (pllim) [inline CHANGES.rst:101]:
```suggestion

- Check unit type (e.g., flux density, surface brightness, counts, etc) for generating
```

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 4]
```diff
@@ -545,7 +545,24 @@ def _plot_uncertainties(self):
...
+                flux_unit_type = "Surface brightness"
+            elif y_unit.is_equivalent(u.erg / (u.s * u.cm**2 * u.Hz)):
+                flux_unit_type = 'Flux'
+            elif y_unit.is_equivalent(u.electron / u.s):
+                flux_unit_type = "Counts"
+            elif y_unit.is_equivalent(u.W):
+                flux_unit_type = "Luminosity"
+
+        # Set x axes labels for the spectrum viewer
         flux_unit_type = "Flux density"
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:566]:
```suggestion
```

**Review by pllim — COMMENTED** [review]
I think it is also a matter of how consistent we want to be w.r.t. `y_unit.physical_type` that you get from `astropy.units`. If consistency isn't a concern, then what you have is fine.

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 5]
```diff
@@ -545,8 +545,25 @@ def _plot_uncertainties(self):
...
+        # Set y axes labels for the spectrum viewer
+        y_display_unit = self.state.y_display_unit
+        y_unit = u.Unit(y_display_unit) if y_display_unit else u.dimensionless_unscaled
+
+        # default to Flux Density for flux density or uncaught types
         flux_unit_type = "Flux density"
+
+        if y_unit.is_equivalent(u.Jy):
+            flux_unit_type = "Flux density"
+        elif y_unit.is_equivalent(u.Jy / u.sr):
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:557]: Given L553, the first `if` using `u.Jy` isn't needed?
```suggestion
        if y_unit.is_equivalent(u.Jy / u.sr):
```

**Review by pllim — COMMENTED** [review]
Do we need tests?

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 6]
```diff
@@ -545,8 +545,23 @@ def _plot_uncertainties(self):
                 self.figure.marks = list(self.figure.marks) + [error_line_mark]
 
     def set_plot_axes(self):
-        # Set axes labels for the spectrum viewer
+        # Set y axes labels for the spectrum viewer
+        y_display_unit = self.state.y_display_unit
+        y_unit = u.Unit(y_display_unit) if y_display_unit else u.dimensionless_unscaled
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:550]: Should `u.dimensionless_unscaled` be dumped into "Counts" category?

p.s. I thought I added this as part of the "review" but looks like it didn't go in.

**Comment by cshanahan1** [comment]
> Do we need tests?

i'm not sure how to test this without adding a new attribute, the label is not accessible

**Comment by pllim** [comment]
I think you can access `self.figure.axes[1].label` and check that the desired flux unit type is there, no?

**Comment by cshanahan1** [comment]
added tests

File: CHANGES.rst [hunk 7]
```diff
@@ -99,6 +99,10 @@ Mosviz
 Specviz
 ^^^^^^^
 
+- Check unit type (e.g flux density, surface brightness, counts, etc) for generating
+  display label for the y axis in spectral viewer. Previously it was hard coded
+  to always display `flux density` no matter the input unit. [#2703]
```
Comment (pllim) [inline CHANGES.rst:104]:
```suggestion
  to always display ``flux density`` no matter the input unit. [#2703]
```

File: jdaviz/configs/specviz/tests/test_viewers.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import numpy as np
+import pytest
+from specutils import Spectrum1D
+
+
+@pytest.mark.parametrize(
+    ('input_unit', 'y_axis_label'),
+    [(u.MJy, 'Flux density'),
+     (u.MJy / u.sr, 'Surface brightness'),
+     (u.electron / u.s, 'Counts'),
```
Comment (pllim) [inline jdaviz/configs/specviz/tests/test_viewers.py:11]:
```suggestion
     (u.electron / u.s, 'Counts'),
     (u.dimensionless_unscaled, 'Counts'),
```

File: jdaviz/configs/specviz/tests/test_viewers.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+     (u.erg / (u.s * u.cm ** 2), 'Flux'),
+     (u.erg / u.s, 'Luminosity')])
+def test_spectrum_viewer_axis_labels(specviz_helper, input_unit, y_axis_label):
+
+    flux = np.arange(1, 10) * input_unit
+    spectral_axis = np.arange(1, 10)
+
+    spec = Spectrum1D(flux, spectral_axis * u.nm)
```
Comment (pllim) [inline jdaviz/configs/specviz/tests/test_viewers.py:19]: Just to be more consistent.
```suggestion
    spectral_axis = np.arange(1, 10) * u.um

    spec = Spectrum1D(flux, spectral_axis)
```

File: jdaviz/configs/specviz/tests/test_viewers.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+    specviz_helper.load_data(spec)
+
+    label = specviz_helper.app.get_viewer_by_id('specviz-0').figure.axes[1].label
+    label = label.split('[')[0].strip()
+
+    assert (label == y_axis_label)
```
Comment (pllim) [inline jdaviz/configs/specviz/tests/test_viewers.py:27]: Maybe this is enough?

```suggestion
    assert y_axis_label in label
```

**Comment by pllim** [comment]
Thanks! Won't catch edge cases like `u.electron` (without per second) but we can handle those as need arises because there are unlimited combos possible.

File: CHANGES.rst [hunk 11]
```diff
@@ -99,6 +99,10 @@ Mosviz
 Specviz
 ^^^^^^^
 
+- Check unit type (e.g flux density, surface brightness, counts, etc) for generating
```
Comment (pllim) [inline CHANGES.rst:102]:
```suggestion
- Check unit type (e.g., flux density, surface brightness, counts, etc) for generating
```

**Comment by pllim** [comment]
devdeps failure is unrelated (https://github.com/glue-viz/glue-jupyter/issues/421)

File: jdaviz/configs/specviz/plugins/viewers.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/specviz/plugins/viewers.py:561]: given that the unit is also in the label, I think this is probably ok?

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 12]
```diff
@@ -545,8 +545,24 @@ def _plot_uncertainties(self):
...
+
+        # default to Flux Density for flux density or uncaught types
         flux_unit_type = "Flux density"
```
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/viewers.py:553]: this might be a little easier to read (and catch bugs like previously happened), if this was in an else at the end of the elifs 🤷‍♂️

**Review by kecnry — APPROVED** [review]
Definitely an improvement and we can always iterate or move to somewhere more general (for all axes in all viewers) down the road.  Thanks for finding and fixing this!

File: jdaviz/configs/specviz/plugins/viewers.py (hunk 12 above)
Comment (kecnry) [reply jdaviz/configs/specviz/plugins/viewers.py]: but it needs to be in an else, otherwise it will always be applied

Comment (cshanahan1) [reply jdaviz/configs/specviz/plugins/viewers.py]: oh whoops haha

**Review by pllim — APPROVED** [review]
LGTM. Thanks!
